rueidis.go

```diff
@@ -267,6 +269,11 @@ func NewClient(option ClientOption) (client Client, err error) {
 	if option.Dialer.KeepAlive == 0 {
 		option.Dialer.KeepAlive = DefaultTCPKeepAlive
 	}
+	// Check if NoTouch is enabled
+	if option.NoTouch {
+		option.PipelineMultiplex = singleClientMultiplex(option.PipelineMultiplex)
+		return newSingleClient(&option, nil, makeConn)
+	}
```

Comment:
These lines look wired to me. I am much expecting to modify here: https://github.com/redis/rueidis/blob/main/pipe.go#L133 and here: https://github.com/redis/rueidis/blob/main/pipe.go#L194